client/src/snippets.json

```diff
@@ -24,17 +38,114 @@
         "prefix": "gx-conditional-select",
         "body": [
             "<conditional name=\"$1\">",
-            "    <param name=\"$2\" type=\"select\" label=\"$3\">",
-            "        <option value=\"$4\">TODO: option $4</option>",
+            "    <param name=\"$2\" type=\"select\" label=\"$3\" help=\"$4\">",
             "        <option value=\"$5\">TODO: option $5</option>",
+            "        <option value=\"$6\">TODO: option $6</option>",
             "    </param>",
-            "    <when value=\"$4\">",
+            "    <when value=\"$5\">",
             "        $0",
             "    </when>",
-            "    <when value=\"$5\">",
+            "    <when value=\"$6\">",
             "    </when>",
             "</conditional>"
         ],
         "description": "Conditional parameter select with two options"
+    },
+    "Select parameter with 3 options": {
+        "prefix": "gx-select-3-param",
+        "body": [
+            "<param name=\"$1\" type=\"select\" label=\"$2\" help=\"$3\">",
+            "    <option value=\"$4\">TODO: option $4</option>",
+            "    <option value=\"$5\">TODO: option $5</option>",
+            "    <option value=\"$6\">TODO: option $6</option>",
+            "</param>"
+        ],
+        "description": "Select parameter with 3 options"
+    },
+    "Select parameter with 5 options": {
+        "prefix": "gx-select-5-param",
+        "body": [
+            "<param name=\"$1\" type=\"select\" label=\"$2\" help=\"$3\">",
+            "    <option value=\"$4\">TODO: option $4</option>",
+            "    <option value=\"$5\">TODO: option $5</option>",
+            "    <option value=\"$6\">TODO: option $6</option>",
+            "    <option value=\"$7\">TODO: option $7</option>",
+            "    <option value=\"$8\">TODO: option $8</option>",
+            "</param>"
+        ],
+        "description": "Select parameter with 5 options"
+    },
+    "Boolean parameter": {
+        "prefix": "gx-boolean-param",
+        "body": [
+            "<param argument=\"$1\" type=\"boolean\" truevalue=\"$2\" falsevalue=\"\" label=\"$3\" help=\"$4\" />"
+        ],
+        "description": "Boolean parameter"
+    },
+    "Integer parameter": {
+        "prefix": "gx-integer-param",
+        "body": [
+            "<param argument=\"$1\" type=\"integer\" min=\"$2\" max=\"$3\" value=\"$4\" label=\"$5\" help=\"$6\" />"
+        ],
+        "description": "Integer parameter"
+    },
+    "Float parameter": {
+        "prefix": "gx-float-param",
+        "body": [
+            "<param argument=\"$1\" type=\"float\" min=\"$2\" max=\"$3\" value=\"$4\" label=\"$5\" help=\"$6\" />"
+        ],
+        "description": "Float parameter"
+    },
+    "Data parameter with single required dataset": {
+        "prefix": "gx-data-single-param",
+        "body": [
+            "<param argument=\"$1\" name==\"$1\" type=\"data\" format==\"$2\" label=\"$3\" help=\"$4\" />"
```

Comment:
```suggestion
            "<param argument=\"--$1\" name=\"$1\" type=\"data\" format=\"$2\" label=\"$3\" help=\"$4\" />"
```

Reply:
The `--` I'm not sure ... it could also be a single `-`

Reply:
Ok, we can remove the `--` but in that case, we should also remove `name="$1"` because it will reuse the $1 place-holder resulting in `name="--sample"` when it should be `name="sample"` isn't it?

Reply:
Yeah, I was not sure how to handle this. I just add name in this example because most tools do not have classical arguments for the input/output files but use positional arguments ... 

I could remove arguments here, but in the cases where an argument could be used, it would be useful ... not sure.